### PR TITLE
docs: Specify that token-2022 is ready and eligible for bounties

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

Token-2022 has finished its audits for version 1.0.0, making it eligible for bug bounties, but the documentation is out of date.

#### Solution

Say that token-2022 is ready!

### CODE REVIEW

File: docs/src/token-2022/status.md [hunk 1]
```diff
@@ -2,17 +2,18 @@
 title: Project Status
 ---
 
-The Token-2022 program is still under audit and not meant for full production use.
-In the meantime, all clusters have the latest program deployed **for testing and
-development purposes ONLY**.
+The Token-2022 program is ready for full production use. All clusters have the
+latest program deployed **without confidential transfer functionality**.
+
```
Comment (t-nelson) [inline docs/src/token-2022/status.md:7]: this seems more like a temporary "announcement" than something that needs to live in the docs?

File: docs/src/token-2022.md [hunk 2]
```diff
@@ -8,9 +8,12 @@ fungible and non-fungible tokens.
...
 [Token Program](token.mdx).
 
-The program is still under audit and not meant for full production use. In the
-meantime, all clusters have the latest program deployed **for testing and development
-purposes ONLY**.
+The program is ready for full production use. All clusters have the latest
+program deployed **without confidential transfer functionality**.
+
+The program with confidential transfer functionality will be deployed once
+Solana v1.17 reaches mainnet-beta with the appropriate syscalls enabled. See the
```
Comment (t-nelson) [inline docs/src/token-2022.md:15]: replicode? maybe just replace with a spruced up link out to the status?

File: docs/src/token-2022.md [hunk 3]
```diff
@@ -138,8 +141,8 @@ For information about the types and instructions, the Rust docs are available at
 
 ## Security Audits
 
-The Token-2022 Program is currently under multiple audits to ensure safety of
-funds. All audits will be published here as they are completed.
+The Token-2022 Program has been audited multiple times to ensure safety of
```
Comment (t-nelson) [inline docs/src/token-2022.md:144]: are the lawyers cool with this language? audits don't "ensure" anything. :trollface:

Comment (joncinque) [reply docs/src/token-2022.md:144]: Oof yeah, that's a good point, I'll just say it's been audited multiple times

File: docs/src/token-2022.md (hunk 2 above)
Comment (joncinque) [reply docs/src/token-2022.md:15]: That's a good point... We had it here as extra cover-your-ass to be really clear that it wasn't ready for prod. we could just remove the whole thing, what do you think?

File: docs/src/token-2022/status.md (hunk 1 above)
Comment (joncinque) [reply docs/src/token-2022/status.md:7]: How about I get rid of the first sentence? At least knowing what's deployed should be pretty useful

**Review by t-nelson — APPROVED** [review]

**Pull request merged**